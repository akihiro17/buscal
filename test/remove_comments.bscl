fn clean_file(path: str) -> void {
  var before: str = capture(["grep", "-c", "^", path]);
  echo(format("The file {} starts with {}", path, before));

  execute(["sed", unquoted_str("'\\/^\\s*#\/d;\/^$/d'"), path]);
}

var file_name: str = "test/remove_comments.input";
clean_file(file_name);

