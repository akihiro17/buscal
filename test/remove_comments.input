# 一行目
吾輩は猫である。名前はまだ無い。
# second line
どこで生れたかとんと見当がつかぬ
# end
