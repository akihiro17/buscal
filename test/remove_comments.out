The file test/remove_comments.input starts with 5
吾輩は猫である。名前はまだ無い。
どこで生れたかとんと見当がつかぬ
